content/developers/developer-patterns/navigating-merklelogs/index.md

```diff
@@ -267,56 +223,41 @@ h.update(bytes.fromhex("018e84dbbb6513a6"[2:]))
 h.update("assets/31de2eb6-de4f-4e5a-9635-38f7cd5a0fc8/events/21d55b73-b4bc-4098-baf7-336ddee4f2f2".encode())
 h.hexdigest()
 ```
-The variable portion *for
-the first massif* contains exactly *16383* MMR *nodes*. Of those nodes, *8192*
-are the leaf entries in the merkle tree corresponding to your events.
-
+The variable portion *for the first massif* contains exactly *16383* MMR *nodes*. Of those nodes, *8192* are the leaf entries in the Merkle tree corresponding to your events.
 
-When a massif is initialised the trieData is pre populated for all leaves and
-set to all zero bytes. As events are recorded in the log, the zero padded index
-is filled in. A sub range of field 0 will change when we save the last
-idtimestamp in it. The mmr node values are strictly only ever appended to the
-blob. Once appended they will never change and they will never move.
+When a massif is initialized the trieData is pre populated for all leaves and set to all zero bytes. As events are recorded in the log, the zero padded index is filled in. A sub range of field 0 will change when we save the last idtimestamp in it. The mmr node values are strictly only ever appended to the blob. Once appended they will never change and they will never move.
 
-If you know the byte offset in the blob for the start of the mmr data then you
-can check the number of mmr nodes currently in it by doing `(blobSize -
-mmrDataStart)/32`.
+If you know the byte offset in the blob for the start of the mmr data then you can check the number of mmr nodes currently in it by doing `(blobSize - mmrDataStart)/32`.
 
 ## The peak stack and mmr data sizes are computable.
 
-Please see [Massif Blob Offset Tables](/developers/developer-patterns/massif-blob-offset-tables)
-if you want to avoid needing to calculate these. Implementations of the O(log base 2 n)
-algorithms are provided in various languages. They all have very hardware
-sympathetic implementations.
+Please see [Massif Blob Pre-Calculated Offsets](/developers/developer-patterns/massif-blob-offset-tables) if you want to avoid needing to calculate these. Implementations of the O(log base 2 n) algorithms are provided in various languages. They all have very hardware sympathetic implementations.
```

Comment:
```suggestion
See [Massif Blob Pre-Calculated Offsets](/developers/developer-patterns/massif-blob-offset-tables) to avoid needing to calculate these.
Implementations of the O(log base 2 n) algorithms are provided in various languages.
They all have very hardware-sympathetic implementations.
```